src/main/java/org/opensearch/knn/index/codec/KNN9120Codec/DerivedSourceStoredFieldsWriter.java

```diff
@@ -65,6 +65,10 @@ public void writeField(FieldInfo info, DataInput value, int length) throws IOExc
 
     @Override
     public int merge(MergeState mergeState) throws IOException {
+        // We have to wrap these here to avoid storing the vectors during merge
+        for (int i = 0; i < mergeState.storedFieldsReaders.length; i++) {
```

Comment:
Do we need null check on mergeState?

Reply:
the iteration is based on the segments, we don't need to do a null check here.